src/TimeManager.jl

```diff
@@ -48,4 +49,18 @@ datetime_to_strdate(datetime::DateTime) =
     string(string(lpad(Dates.month(datetime), 2, "0"))) *
     string(lpad(Dates.day(datetime), 2, "0"))
 
+abstract type AbstractFrequency end
+struct Monthly <: AbstractFrequency end
+struct EveryTimestep <: AbstractFrequency end
+
+"""
+    trigger_callback(cs, ::Monthly)
+    
+Returns `true` if the current date is equal to or exceeds the saved first of the month at time of 00:00:00. 
+
+# Arguments
+- `cs`: [CoupledSimulation] containing info about the simulation
+"""
+trigger_callback(cs::CoupledSimulation, ::Monthly) = cs.dates.date[1] >= cs.dates.date1[1] ? true : false
```

Comment:
why does this need the `::Monthly` param?

Reply:
Right now it's more like a label for clarity. But in the future, we may want to have callbacks for `Daily` or `EveryTimestep`. This callback interface will probably change though once we start thinking about more complex timestepping.